spinalcordtoolbox/gui/centerline.py

```diff
@@ -31,13 +32,15 @@ def reformat_image(self):
         # reorient data to SAL
         super(CenterlineController, self).reformat_image()
         max_x, max_z = self.image.dim[:3:2]
-        self.params.num_points = self.params.num_points or 11
+        # self.params.num_points = self.params.num_points or 11
```

Comment:
there might be legacy code in the function that refer to this line-- i didn't want to remove it completely in case there was still dependent code, and i would need more time (which i don't have) to make sure of that.

Reply:
In that case, we should probably investigate now, or else we would be introducing a bug into v5.1.0 by rushing things. :sweat_smile: 

I can look into this today.

Reply:
that's wise. Thanks!

Reply:
This LOC was originally added in https://github.com/neuropoly/spinalcordtoolbox/pull/1496, specifically by e3f5c6231f1e11eb8d566126e9e393280484fc1e. The change from that commit was this:

```python
# old 
self.INTERVAL = int(10 / self.image.dim[4]) or 1
if not self.params.num_points:
    self.params.num_points = self.image.dim[0] / self.INTERVAL

# new
self.params.num_points = self.params.num_points or 11
self.INTERVAL = max_z // (self.params.num_points - 1) or 3 
```

So, `11` was an attempt to set a new sensible default value for `num_points` in the auto mode of `CenterlineController`. 

By commenting out this line, the default instead comes from the default value from `AnatomicalParams()`:

https://github.com/neuropoly/spinalcordtoolbox/blob/7efbbb2db90afdb577f836958413bb70fd29c00e/spinalcordtoolbox/gui/base.py#L62

Reply:
However, in terms of sct scripts, the default doesn't even matter because we override `params.num_points` wherever we call `launch_centerline_dialog` anyway:

https://github.com/neuropoly/spinalcordtoolbox/blob/7efbbb2db90afdb577f836958413bb70fd29c00e/spinalcordtoolbox/scripts/sct_propseg.py#L569-L578

https://github.com/neuropoly/spinalcordtoolbox/blob/25bbc2ec0d40969c7ed37e2a5d56d4deb7a1dfc1/spinalcordtoolbox/centerline/core.py#L285-L289

Reply:
So.... in summary, I think this change doesn't _really_ have an effect, but the default values probably should be cleaned up at some point.